### PR TITLE
EXPERIMENTAL: Zephyr 3.4 update

### PR DESCRIPTION
**Pull request by petejohanson**
This is a placeholder for now, so folks who have reported BLE issues on Windows in particular can test latest Zephyr BLE stack and see if this improves things.

This has *not* gotten extensive testing yet, and should be considered *highly* experimental.

### CODE REVIEW

File: app/src/wpm.c [hunk 1]
```diff
@@ -72,7 +72,7 @@ void wpm_expiry_function() { k_work_submit(&wpm_work); }
 
 K_TIMER_DEFINE(wpm_timer, wpm_expiry_function, NULL);
 
-int wpm_init() {
+static int wpm_init(void) {
```
Comment (mike1808) [inline app/src/wpm.c:75]: Why do this become static?

Comment (petejohanson) [reply app/src/wpm.c:75]: For consistency, it should have been static in the first place, it doesn't need to be an exported public symbol.

**Comment by mike1808** [comment]
flashed my keyboard with it, the only issue i noticed is that my central side sharp memory display was not working, no errors in the log. I'll try to see why they don't work later.

File: app/src/activity.c [hunk 2]
```diff
@@ -74,8 +74,8 @@ void activity_work_handler(struct k_work *work) {
     } else
 #endif /* IS_ENABLED(CONFIG_ZMK_SLEEP) */
         if (inactive_time > MAX_IDLE_MS) {
-            set_state(ZMK_ACTIVITY_IDLE);
-        }
+        set_state(ZMK_ACTIVITY_IDLE);
```
Comment (chrissicool) [inline app/src/activity.c:77]: The indentation looks messed up here.

**Comment by szczys** [comment]
I think I found two more LV_Z_DPI symbols that can be changed: 

```
diff --git a/app/boards/shields/leeloo_micro/Kconfig.defconfig b/app/boards/shields/leeloo_micro/Kconfig.defconfig
index 26256120..009aca55 100644
--- a/app/boards/shields/leeloo_micro/Kconfig.defconfig
+++ b/app/boards/shields/leeloo_micro/Kconfig.defconfig
@@ -34,7 +34,7 @@ if LVGL
 config LV_Z_VDB_SIZE
     default 64

-config LV_Z_DPI
+config LV_DPI_DEF
     default 148

 config LV_Z_BITS_PER_PIXEL
diff --git a/app/boards/shields/nice_view/Kconfig.defconfig b/app/boards/shields/nice_view/Kconfig.defconfig
index 53edc1cc..84fdd7cc 100644
--- a/app/boards/shields/nice_view/Kconfig.defconfig
+++ b/app/boards/shields/nice_view/Kconfig.defconfig
@@ -6,7 +6,7 @@ if SHIELD_NICE_VIEW
 config LV_Z_VDB_SIZE
     default 100

-config LV_Z_DPI
+config LV_DPI_DEF
     default 161

 config LV_Z_BITS_PER_PIXEL
```

**Comment by xudongzheng** [comment]
Note for Zephyr 3.5:

Previously SPI3 could be defined with the MOSI pin and nothing else when configuring WS2812B. Zephyr 3.5 seems to require a dummy SCK pin on SPI3 (just like it was necessary for SPI0/SPI1/SPI2).

**Comment by petejohanson** [comment]
Superseded by #1995

**Pull request closed without merging**